src/main/kotlin/net/leanix/githubagent/services/WebhookService.kt

```diff
@@ -0,0 +1,85 @@
+    private val cachingService: CachingService,
+    private val gitHubAuthenticationService: GitHubAuthenticationService
+) {
+
+    private val logger = LoggerFactory.getLogger(WebhookService::class.java)
+    private val objectMapper = jacksonObjectMapper()
+
+    fun consumeWebhookEvent(eventType: String, payload: String) {
+        when (eventType.uppercase()) {
+            "PUSH" -> handlePushEvent(payload)
+            else -> {
+                logger.info("Sending event")
+                webSocketService.sendMessage("/events/other", payload)
+            }
+        }
+    }
+
+    private fun handlePushEvent(payload: String) {
+        val pushEventPayload: PushEventPayload = objectMapper.readValue(payload)
+        val repositoryName = pushEventPayload.repository.name
+        val repositoryFullName = pushEventPayload.repository.fullName
+        val headCommit = pushEventPayload.headCommit
+        val organizationName = pushEventPayload.repository.owner.name
+
+        var installationToken = cachingService.get("installationToken:${pushEventPayload.installation.id}")?.toString()
+        if (installationToken == null) {
+            gitHubAuthenticationService.refreshTokens()
+            installationToken = cachingService.get("installationToken:${pushEventPayload.installation.id}")?.toString()
+            require(installationToken != null) { "Installation token not found/ expired" }
+        }
+
+        if (pushEventPayload.ref == "refs/heads/${pushEventPayload.repository.defaultBranch}") {
+            when {
+                MANIFEST_FILE_NAME in headCommit.added -> {
+                    logger.info("Manifest file added to repository $repositoryFullName")
+                    val fileContent = getFileContent(organizationName, repositoryName, installationToken)
+                    sendManifestData(repositoryFullName, ManifestFileAction.ADDED, fileContent)
+                }
+                MANIFEST_FILE_NAME in headCommit.modified -> {
+                    logger.info("Manifest file modified in repository $repositoryFullName")
+                    val fileContent = getFileContent(organizationName, repositoryName, installationToken)
+                    sendManifestData(repositoryFullName, ManifestFileAction.MODIFIED, fileContent)
+                }
+                MANIFEST_FILE_NAME in headCommit.removed -> {
+                    logger.info("Manifest file removed from repository $repositoryFullName")
+                    sendManifestData(repositoryFullName, ManifestFileAction.REMOVED, null)
+                }
+            }
+        }
+    }
+
+    private fun getFileContent(organizationName: String, repositoryName: String, token: String): String {
+        return gitHubGraphQLService.getFileContent(
+            owner = organizationName,
+            repositoryName,
+            "HEAD:${gitHubEnterpriseProperties.manifestFileDirectory}$MANIFEST_FILE_NAME",
+            token
+        )
+    }
```

Comment:
I don't think this function is needed. `gitHubGraphQLService.getFileContent()` can be called directly